ext/nnstreamer/tensor_filter/tensor_filter_caffe2_core.cc

```diff
@@ -300,7 +303,27 @@ Caffe2Core::run (const GstTensorMemory * input, GstTensorMemory * output)
     }
   }
 
-  workSpace.RunNet (predictNet.name ());
+  /**
+   * As the input information has not been verified, the first run for the model
+   * is encapsulated in a try-catch block
+   */
+  if (first_run) {
+    try {
+      workSpace.RunNet (predictNet.name ());
+      first_run = false;
+    } catch(const std::runtime_error& re) {
+      g_critical ("Runtime error while running the model: %s", re.what());
+      return -4;
+    } catch(const std::exception& ex)	{
+      g_critical ("Exception while running the model : %s", ex.what());
+      return -4;
+    } catch (...) {
+      g_critical ("Unknown exception while running the model");
+      return -4;
+    }
+  } else {
+    workSpace.RunNet (predictNet.name ());
+  }
```

Comment:
Does it have some problems to wrap `workspace.RunNet()` with try/catch in next runs?

Reply:
In my understanding, wrapping with try/catch block adds overhead, and will make the execution slower.

Reply:
Well.. this overhead is very little because running a network already takes a few milliseconds.
We can ignore the costs in most cases because most overheads come from throwing an exception and handling it. 

So, IMO, next runs may also have a possibility to generate runtime errors.
How about just wrapping try/catch for next runs as well?

Reply:
The purpose of this try catch is to only catch errors related to mismatch of input/output.
I didn't know that the overhead of try/catch is very small when exception handling is not very frequent.
Yes, wrapping for all runs makes sense as well.
Then I think this should be done for all the tensor filter extensions?

Reply:
Oh, if its purpose is just to check the mismatch of input/output, not runtime errors, I think it's okay.